### PR TITLE
{Script} Add retry in get_whl_from_url and only verify latest version of an extension

### PR DESCRIPTION
**Pull request by fengzhou-msft**
To mitigate the increasing failures in `Verify Extensions Index` CI task recently.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Script

File: scripts/ci/test_index.py [hunk 1]
```diff
@@ -21,6 +21,10 @@
 
 from util import get_ext_metadata, get_whl_from_url, get_index_data, verify_dependency
 
+# Some extensions are published through manually submitted PRs and the latest version may not be in the end of
+# the version list in index.json. Let's just test all versions for these extensions.
+EXTS_TEST_ALL_VERSIONS = {'azure-cli-ml', 'alias', 'azure_iot'}
```
Comment (jiasli) [inline scripts/ci/test_index.py:26]: Maybe we can sort all versions and only test the latest one?

File: scripts/ci/test_index.py [hunk 2]
```diff
@@ -105,8 +109,12 @@ def test_filename_duplicates(self):
 
     @unittest.skipUnless(os.getenv('CI'), 'Skipped as not running on CI')
     def test_checksums(self):
-        for exts in self.index['extensions'].values():
-            for item in exts:
+        for ext_name, exts in self.index['extensions'].items():
+            end = len(exts) - 1
+            for index, item in enumerate(exts):
```
Comment (jiasli) [inline scripts/ci/test_index.py:114]: Redefine the name `index` can be confusing to readers. 😉

File: scripts/ci/test_index.py (hunk 1 above)
Comment (jiasli) [reply scripts/ci/test_index.py:26]: Python's built-in function [`max`](https://docs.python.org/3/library/functions.html#max) supports a custom key extraction function and we can use [`packaging.version`](https://packaging.pypa.io/en/latest/version.html) for version comparison.

Comment (fengzhou-msft) [reply scripts/ci/test_index.py]: updated with picking a `max` version so the order of versions does not matter now.

File: scripts/ci/test_index.py [hunk 3]
```diff
@@ -106,14 +106,16 @@ def test_filename_duplicates(self):
...
-            for item in exts:
-                ext_file = get_whl_from_url(item['downloadUrl'], item['filename'],
-                                            self.whl_cache_dir, self.whl_cache)
-                computed_hash = get_sha256sum(ext_file)
-                self.assertEqual(computed_hash, item['sha256Digest'],
-                                 "Computed {} but found {} in index for {}".format(computed_hash,
-                                                                                   item['sha256Digest'],
-                                                                                   item['filename']))
+            # only test the latest version
+            item = max(exts, key=lambda ext: LooseVersion(ext['metadata']['version']))
```
Comment (jiasli) [inline scripts/ci/test_index.py:110]: `distutils.version.LooseVersion` is legacy and conformant only to the superseded [PEP 386](https://www.python.org/dev/peps/pep-0386/) (by [PEP 440](https://www.python.org/dev/peps/pep-0440/)). Perhaps we can migrate to the latest [`packaging.version`](https://packaging.pypa.io/en/latest/version.html).

See https://stackoverflow.com/a/11887885/2199657

Comment (jiasli) [reply scripts/ci/test_index.py:110]:
```py
>>> from distutils.version import LooseVersion, StrictVersion
>>> StrictVersion('1.0.0a1') < StrictVersion('1.0.0')
True
>>> LooseVersion('1.0.0a1') < LooseVersion('1.0.0')
False
```

File: scripts/ci/test_index.py [hunk 4]
```diff
@@ -16,7 +16,7 @@
 import hashlib
 import shutil
 
-from distutils.version import LooseVersion
+from packaging import version
```
Comment (fengzhou-msft) [inline scripts/ci/test_index.py:19]: `distutils.version.LooseVersion` does not support PEP440 which we follow as discussed in https://stackoverflow.com/a/11887885/2199657

**Review by jiasli — APPROVED** [review]

**Pull request merged**